### DISCUSSION TITLE
Authorize the application with your OneDrive Account

### DISCUSSION THREAD

**Discussion by sweng5080** [abraunegg/onedrive#2237] [Q&A] [answered]
While trying to get OneDrive Business to authorize and following the [USAGE](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md) documentation  **Authorize the application with your OneDrive Account** it is not obvious what the name of the "app" that should be authorized. When requesting a new app from the IT department, what name explicitly should be requested if its not listed?

**Comment by abraunegg** [comment, marked as answer]
@sweng5080 
> When requesting a new app from the IT department, what name explicitly should be requested if its not listed?

You can use any name you choose .. this is a 'logical' name to refer to the application GUID that your IT Team will create.

**Comment by sweng5080** [comment]
I still don't follow what explicitly needs to be requested from the IT department.
This is the result of following the "Authorize this app visiting: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=..." link in the browser after a few clicks with username and password information:
![image](https://user-images.githubusercontent.com/3439065/203445465-4612f6b4-d285-468e-a495-89edcf04c87c.png)

**Reply by abraunegg** [reply]
> I still don't follow what explicitly needs to be requested from the IT department.

You are *not* an administrator for the Active Directory | Azure authentication domain associated with your Microsoft account. This is why you see this message.

You need to seek assistance from the admin folk who administer your domain.

The message you are presented with is one that Microsoft created|generates - this is a Microsoft authentication requirement and this step of the process has nothing to do with this application, other than providing the 'application GUID' that you are requesting to be authorised.

Please speak to whomever manages your network.
